aiida/schedulers/plugins/direct.py

```diff
@@ -142,17 +142,8 @@ def _get_submit_script_header(self, job_tmpl):
                 lines.append('exec 2>&1')
 
         if job_tmpl.max_memory_kb:
-            try:
-                virtual_memory_kb = int(job_tmpl.max_memory_kb)
-                if virtual_memory_kb <= 0:
-                    raise ValueError
-            except ValueError:
-                raise ValueError(
-                    'max_memory_kb must be '
-                    "a positive integer (in kB)! It is instead '{}'"
-                    ''.format((job_tmpl.max_memory_kb))
-                )
-            lines.append(f'ulimit -v {virtual_memory_kb}')
+            self.logger.info('Physical memory limiting is not supported by the direct scheduler.')
```

Comment:
I think this should be a warning - the user is requesting something that is not supported.